### PR TITLE
feat: add subgraph health endpoint 

### PR DESCRIPTION
**Pull request by shiyasmohd**
This PR is the second part of #408

### CODE REVIEW

File: common/src/indexer_service/http/indexer_service.rs [hunk 1]
```diff
@@ -404,6 +404,11 @@ impl IndexerService {
             ),
         };
 
+        // Check subgraph Health
+        misc_routes = misc_routes
+            .route("/subgraph/health/:deployment_id", get(health))
+            .route_layer(Extension(options.config.graph_node.clone()));
+
```
Comment (gusinacio) [inline common/src/indexer_service/http/indexer_service.rs:411]: This is missing a `rate_limiter`. We mustn't overwhelm graph-node.

File: common/src/indexer_service/http/health.rs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                        message
+                    }
+                    nonFatalErrors {
+                        message
+                    }
+                }
+            }
+        "#,
+        [("ids", json!([deployment_id]))],
+    );
```
Comment (gusinacio) [inline common/src/indexer_service/http/health.rs:102]: Could you use `graphql-client` for this? You maybe need to add the schema for it too.

**Review by gusinacio — CHANGES_REQUESTED** [review]
Hello, thanks for the code. These are some concerns regarding health rate limit and using a different tool for graphql queries.

**Comment by shiyasmohd** [comment]
@gusinacio Thanks for the review. Pushed the changes.

File: common/src/indexer_service/http/health.rs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    });
+
+    let client = reqwest::Client::new();
+    let response = client
+        .post(graph_node.status_url)
+        .json(&req_body)
+        .send()
+        .await;
+    let res = response.expect("Failed to get response");
+    let response_json: Result<Response, reqwest::Error> = res.json().await;
```
Comment (gusinacio) [inline common/src/indexer_service/http/health.rs:107]: Hey, graphql-client already has all the types for you. You can just use `health_query::ResponseData`

**Review by gusinacio — CHANGES_REQUESTED** [review]

File: common/src/indexer_service/http/health.rs (hunk 3 above)
Comment (shiyasmohd) [reply common/src/indexer_service/http/health.rs]: Hey @gusinacio 
Thanks the suggestion. Pushed the changes. Let me know if any changes needed.

File: common/src/indexer_service/http/health.rs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        (Some(data), None) => data,
+        (_, Some(_)) => return Err(CheckHealthError::BadResponse),
+        (_, _) => return Err(CheckHealthError::BadResponse),
+    };
+
+    if data.indexing_statuses.is_empty() {
+        return Err(CheckHealthError::DeploymentNotFound);
+    }
+
+    let status = &data.indexing_statuses[0];
```
Comment (gusinacio) [inline common/src/indexer_service/http/health.rs:90]:
```suggestion
    let Some(status) = data.indexing_statuses.first() else {
        return Err(CheckHealthError::DeploymentNotFound);
    };
```

File: common/src/indexer_service/http/health.rs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    let graphql_response: graphql_client::Response<health_query::ResponseData> = response
+        .json()
+        .await
+        .map_err(|_| CheckHealthError::BadResponse)?;
+
+    let data = match (graphql_response.data, graphql_response.errors) {
+        (Some(data), None) => data,
+        (_, Some(_)) => return Err(CheckHealthError::BadResponse),
+        (_, _) => return Err(CheckHealthError::BadResponse),
```
Comment (gusinacio) [inline common/src/indexer_service/http/health.rs:83]:
```suggestion
        _ => return Err(CheckHealthError::BadResponse),
```

File: common/src/indexer_service/http/health.rs [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "Invalid health status found",
+            ),
+        };
+
+        let body = serde_json::json!({
+            "error": error_message,
+        });
+
+        (status, Json(body)).into_response()
```
Comment (gusinacio) [inline common/src/indexer_service/http/health.rs:55]:
```suggestion
        let status = match &self {
            CheckHealthError::DeploymentNotFound => StatusCode::NOT_FOUND,
            CheckHealthError::InvalidHealthStatus
            | CheckHealthError::BadResponse => StatusCode::INTERNAL_SERVER_ERROR,
            CheckHealthError::RequestFailed => StatusCode::BAD_GATEWAY,
        };

        let body = serde_json::json!({
            "error": self.to_string(),
        });

        (status, Json(body)).into_response()
```

**Review by gusinacio — CHANGES_REQUESTED** [review]
Some syntax improvement and we are good to merge

**Comment by shiyasmohd** [comment]
@gusinacio  Thanks for the suggestions. Pushed the changes.

**Review by gusinacio — APPROVED** [review]
Beautiful code, really nice work Thanks!

**Pull request merged**